lib/tmdb/client.rb

```diff
@@ -104,116 +108,134 @@ def common_actors_between_movies(movie_one_title, movie_two_title)
+
+        tv_credits_url = url_for_person_tv_credits(person_id)
+        tv_credits_data = get_data(tv_credits_url)
 
         OpenStruct.new(
           person_id: person_id,
-          profile: MoviePersonProfile.parse_result(bio_results),
-          movie_credits: MoviePersonCredits.parse_result(movie_credits_results),
-          tv_credits: TVPersonCredits.parse_result(tv_credits_results)
+          profile: MoviePersonProfile.parse_result(person_data),
+          movie_credits: MoviePersonCredits.parse_result(movie_credits_data),
+          tv_credits: TVPersonCredits.parse_result(tv_credits_data)
         )
       end
 
       def tv_actor_appearance_credits(credit_id)
-        data = get_parsed_credit(credit_id)
+        url = url_for_credits_data(credit_id)
+        data = get_data(url)
         TVActorCredit.parse_record(data)
       end
 
       def tv_series_autocomplete(query)
-        data = get_parsed_tv_search_results(query)
+        url = url_for_tv_series_search(query)
+        data = get_data(url)&.dig(:results)
         data.map{ |d| d[:name] }.uniq
       end
 
       def tv_series_search(query)
-        data = get_parsed_tv_search_results(query)
+        url = url_for_tv_series_search(query)
+        data = get_data(url)&.dig(:results)
         TVSeries.parse_search_records(data) if data.present?
       end
 
       def tv_series(series_id)
-        data = get_parsed_tv_series_data(series_id)
+        url = url_for_tv_series_data(series_id)
+        data = get_data(url)
         TVSeries.parse_record(data, series_id)
       end
 
       def tv_season(series:, season_number:)
-        season_data = get_parsed_tv_season_data(series: series, season_number: season_number)
+        url = url_for_tv_season_data(series: series, season_number: season_number)
+        data = get_data(url)
         TVSeason.parse_record(
           series: series,
-          season_data: season_data
+          season_data: data
         )
       end
 
       def tv_episode(series_id:, season_number:, episode_number:)
-        episode_data = get_parsed_tv_episode_data(
+        url = url_for_tv_episode_data(
           series_id: series_id,
           season_number: season_number,
           episode_number: episode_number
         )
-        TVEpisode.parse_record(episode_data)
+        data = get_data(url)
+        TVEpisode.parse_record(data)
       end
 
       private
 
-      def get_parsed_credit(credit_id)
-        url = "#{BASE_URL}/credit/#{credit_id}?api_key=#{API_KEY}"
+      def request(endpoint, params)
+        endpoints = {
+          credits_data: "/credit/#{params[:credit_id]}?api_key=#{API_KEY}",
+          person_data: "/person/#{params[:person_id]}?api_key=#{API_KEY}",
+          person_movie_credits: "/person/#{params[:person_id]}/movie_credits?api_key=#{API_KEY}"
+        }
+        url = "#{BASE_URL}#{endpoints[endpoint]}"
         JSON.parse(open(url).read, symbolize_names: true)
       end
```

Comment:
This is just the start of what would be a single method to replace all of the `url_for...` methods.

Reply:
👍 Yeah, I really like this approach.

Reply:
This evening at Ruby Tuesday, @hayesmp  gave me the idea i think we were reaching for here  ❤️ 🙌  -- make this hash a case statement instead. This way we're only building the url we want instead of building all the urls when building the hash.